modules/pre-commit.nix

```diff
@@ -315,7 +329,31 @@ in
                 echo 1>&2 "    3. add .pre-commit-config.yaml to .gitignore"
               else
                 ln -s ${configFile} .pre-commit-config.yaml
-                pre-commit install
+                # Remove any previously installed hooks (since pre-commit iteslf has no idempotent design)
+                hooks="pre-commit pre-merge-commit pre-push prepare-commit-msg commit-msg post-checkout post-commit"
+                for hook in $hooks; do
+                  pre-commit uninstall -t $hook
+                done
```

Comment:
This is producing errors like this with the bundled version of pre-commit

```
pre-commit uninstalled
usage: pre-commit uninstall [-h] [--color {auto,always,never}] [-c CONFIG]
                            [-t {pre-commit,pre-merge-commit,pre-push,prepare-commit-msg,commit-msg}]
pre-commit uninstall: error: argument -t/--hook-type: invalid choice: 'post-checkout' (choose from 'pre-commit', 'pre-merge-commit', 'pre-push', 'prepare-commit-msg', 'commit-msg')
usage: pre-commit uninstall [-h] [--color {auto,always,never}] [-c CONFIG]
                            [-t {pre-commit,pre-merge-commit,pre-push,prepare-commit-msg,commit-msg}]
pre-commit uninstall: error: argument -t/--hook-type: invalid choice: 'post-commit' (choose from 'pre-commit', 'pre-merge-commit', 'pre-push', 'prepare-commit-msg', 'commit-msg')
pre-commit installed at .git/hooks/pre-commit
```

I don't see a a reason not to uninstall pre-commit entirely before installing. It's also more future proof.

Reply:
I _think_ `pre-commit` doesn't behave as we wised it to behave, see:

```console
$ pre-commit install
pre-commit installed at .git/hooks/pre-commit

$ pre-commit install -t post-checkout
pre-commit installed at .git/hooks/post-checkout

$ pre-commit uninstall
pre-commit uninstalled

# post-checkout is still installed
$ ll .git/hooks/post-checkout
.rwxrwxr-x blaggacao blaggacao 1.4 KB Tue Jul 21 12:02:26 2020  post-checkout

# Only now it's cleaned up
$ pre-commit uninstall -t post-checkout
post-checkout uninstalled
```
The solution is to either bundle another pre-commit or take out `post_*` hooks from the list

Reply:
Updating our pins here is totally fine. We use `niv` for that.

Reply:
This is getting taxing... :wink:
```console
$ niv add pre-commit
Adding package pre-commit
  FATAL: One or more packages failed to update:
  pre-commit: Update failed: Key could not be found: url_template
  CallStack (from HasCallStack):
    error, called at src/Niv/Update.hs:80:25 in niv-0.2.13-4DFkDQryuRMJTtP7ftbwS:Niv.Update
ERROR: ExitFailure 1
```

Reply:
> Updating our pins here is totally fine. We use niv for that.

Oh I had overlooked. Could you check if the last commit gets it right? (I'm new to niv).

Reply:
Looks kinda alright, although I don't know why yours has a `version` attribute. I get no `version` when running

    nix-shell
    niv drop pre-commit
    niv add pre-commit/pre-commit

Did you use the nix-shell in the repo?

Reply:
[`niv add pre-commit/pre-commit -v v2.6.0`](https://github.com/pre-commit/pre-commit/releases/tag/v2.6.0)

I don't think there are guarantees that `pre-commit` master works all the time.

Reply:
Oh no, seems like `-v` doesn't do anything. https://github.com/nmattia/niv/issues/232
I normally use `-b TAG` which works, even though it was named for branches.

Reply:
Another _darknix_ caveat :smirk: